### PR TITLE
refactor release management

### PR DESCRIPTION
**Pull request by xh3b4sd**
This PR tries to improve the internal Giant Swarm releases management. 

```
$ kubectl-gs template cluster --domain="gauss.eu-central-1.aws.gigantic.io" --master-az="eu-central-1a" --name="xh3b4sd" --owner="giantswarm" --region="eu-central-1" --release="v11.4.1"
apiVersion: cluster.x-k8s.io/v1alpha2
kind: Cluster
metadata:
  creationTimestamp: null
  labels:
    cluster-operator.giantswarm.io/version: 2.3.0
    giantswarm.io/cluster: bku53
    giantswarm.io/organization: giantswarm
    release.giantswarm.io/version: v11.4.1
  name: bku53
  namespace: default
spec:
  infrastructureRef:
    apiVersion: infrastructure.giantswarm.io/v1alpha2
    kind: AWSCluster
    name: bku53
    namespace: default
status:
  controlPlaneInitialized: false
  infrastructureReady: false
---
apiVersion: infrastructure.giantswarm.io/v1alpha2
kind: AWSCluster
metadata:
  creationTimestamp: null
  labels:
    aws-operator.giantswarm.io/version: 8.7.1
    giantswarm.io/cluster: bku53
    giantswarm.io/organization: giantswarm
    release.giantswarm.io/version: v11.4.1
  name: bku53
  namespace: default
spec:
  cluster:
    description: xh3b4sd
    dns:
      domain: gauss.eu-central-1.aws.gigantic.io
    kubeProxy: {}
    oidc:
      claims: {}
  provider:
    credentialSecret:
      name: credential-default
      namespace: giantswarm
    master:
      availabilityZone: eu-central-1a
      instanceType: m5.xlarge
    pods:
      externalSNAT: false
    region: eu-central-1
status:
  cluster: {}
  provider:
    network: {}
---
apiVersion: infrastructure.giantswarm.io/v1alpha2
kind: G8sControlPlane
metadata:
  creationTimestamp: null
  labels:
    cluster-operator.giantswarm.io/version: 2.3.0
    giantswarm.io/cluster: bku53
    giantswarm.io/control-plane: 0nzjg
    giantswarm.io/organization: giantswarm
    release.giantswarm.io/version: v11.4.1
  name: 0nzjg
  namespace: default
spec:
  infrastructureRef:
    apiVersion: infrastructure.giantswarm.io/v1alpha2
    kind: AWSControlPlane
    name: 0nzjg
    namespace: default
  replicas: 1
status: {}
---
apiVersion: infrastructure.giantswarm.io/v1alpha2
kind: AWSControlPlane
metadata:
  creationTimestamp: null
  labels:
    aws-operator.giantswarm.io/version: 8.7.1
    giantswarm.io/cluster: bku53
    giantswarm.io/control-plane: 0nzjg
    giantswarm.io/organization: giantswarm
    release.giantswarm.io/version: v11.4.1
  name: 0nzjg
  namespace: default
spec:
  availabilityZones:
  - eu-central-1a
  instanceType: m5.xlarge
```

### CODE REVIEW

File: cmd/template/cluster/flag.go [hunk 1]
```diff
@@ -21,7 +21,6 @@ const (
 	flagDomain       = "domain"
 	flagMasterAZ     = "master-az"
 	flagName         = "name"
-	flagNoCache      = "no-cache"
```
Comment (xh3b4sd) [inline cmd/template/cluster/flag.go:24]: The local file system magic got removed. No local folder sync anymore.

File: cmd/template/cluster/flag.go [hunk 2]
```diff
@@ -127,9 +124,7 @@ func (f *flag) Validate() error {
 
 	var release *gsrelease.GSRelease
 	{
-		c := gsrelease.Config{
-			NoCache: f.NoCache,
-		}
+		c := gsrelease.Config{}
```
Comment (xh3b4sd) [inline cmd/template/cluster/flag.go:127]: The branch name is configurable. Command flags can now easily be added. Not doing this here now though.

File: internal/key/templates.go [hunk 3]
```diff
@@ -1,23 +1,23 @@
 package key
 
 const AppCRTemplate = `
-{{ .UserConfigConfigMapCR -}}
+{{- .UserConfigConfigMapCR -}}
```
Comment (xh3b4sd) [inline internal/key/templates.go:4]: Polished the template spacing. No empty newline on top anymore.

File: pkg/gsrelease/release.go [hunk 4]
```diff
@@ -16,93 +12,52 @@ import (
 )
 
 const (
-	configRelativePath         = ".kube/gs"
-	releasesConfigRelativePath = ".kube/gs/aws.yaml"
-	releasesAWSBaseURL         = "https://raw.githubusercontent.com/giantswarm/releases/master/aws/"
-	releasesConfigURL          = "https://raw.githubusercontent.com/giantswarm/releases/master/aws/kustomization.yaml"
+	defaultBranch = "master"
```
Comment (xh3b4sd) [inline pkg/gsrelease/release.go:15]: When no branch is given then `master` is the default.

File: pkg/gsrelease/release.go [hunk 5]
```diff
@@ -16,93 +12,52 @@ import (
...
-	ReleaseList []string `yaml:"resources"`
+type Config struct {
+	Branch string
 }
 
 type GSRelease struct {
 	releases []Release
 }
 
-type Release struct {
```
Comment (xh3b4sd) [inline pkg/gsrelease/release.go:37]: Moved that code to `type.go`.

File: pkg/gsrelease/release.go [hunk 6]
```diff
@@ -133,80 +88,35 @@ func (r *GSRelease) Validate(version string) bool {
...
 		if err != nil {
-			return microerror.Mask(err)
+			return nil, microerror.Mask(err)
 		}
-		defer out.Close()
+	}
 
-		_, err = io.Copy(out, resp.Body)
+	r := struct {
+		Resources []string `yaml:"resources"`
```
Comment (xh3b4sd) [inline pkg/gsrelease/release.go:107]: Tried to make all of this a bit more consistent.

File: pkg/gsrelease/release.go [hunk 7]
```diff
@@ -133,80 +88,35 @@ func (r *GSRelease) Validate(version string) bool {
...
-	if err != nil {
-		return nil, microerror.Mask(err)
 	}
 
 	var releases []Release
 	{
-		for _, rv := range rl.ReleaseList {
-			resp, err := http.Get(releasesAWSBaseURL + rv + "/release.yaml")
+		for _, v := range r.Resources {
+			resp, err := http.Get(fmt.Sprintf(releasesAWSBaseURLFmt, branch) + v + "/release.yaml")
```
Comment (axbarsan) [inline pkg/gsrelease/release.go:119]: Can we use [strings.Builder](https://golang.google.cn/pkg/strings/#Builder), instead of string concatenation here?

Comment (xh3b4sd) [reply pkg/gsrelease/release.go]: Have another look. I changed the format to make the string building a bit more straight forward. Let me know what you think.

**Pull request merged**